utils/defaulter.go

```diff
@@ -126,3 +131,69 @@ func (d *Defaulter) MustSet(arg interface{}) {
 		panic(err)
 	}
 }
+
+func (d *Defaulter) getEntitySchema(entityType string) (map[string]interface{}, error) {
+	var schema map[string]interface{}
+	schema, err := d.client.Schemas.Get(d.ctx, entityType)
+	if err != nil {
+		return schema, err
+	}
+	return schema, nil
+}
+
+func (d *Defaulter) addEntityDefaults(entityType string, entity interface{}) error {
+	schema, err := d.getEntitySchema(entityType)
+	if err != nil {
+		return fmt.Errorf("retrieve schema for %v from Kong: %v", entityType, err)
+	}
+	return kong.FillEntityDefaults(entity, schema)
+}
+
+func GetKongDefaulterWithClient(ctx context.Context, client *kong.Client) (*Defaulter, error) {
+	d, err := GetKongDefaulter()
+	if err != nil {
+		return nil, err
+	}
+	d.ctx = ctx
+	d.client = client
+
+	route := kong.Route{}
+	if err := d.addEntityDefaults("routes", &route); err != nil {
+		return nil, fmt.Errorf("get defaults for routes: %v", err)
+	}
+	if err := d.Register(&route); err != nil {
+		return nil, fmt.Errorf("registering route with defaulter: %w", err)
+	}
+
+	service := kong.Service{}
+	if err := d.addEntityDefaults("services", &service); err != nil {
+		return nil, fmt.Errorf("get defaults for services: %v", err)
+	}
+	if err := d.Register(&service); err != nil {
+		return nil, fmt.Errorf("registering service with defaulter: %w", err)
+	}
+
+	upstream := kong.Upstream{}
+	if err := d.addEntityDefaults("upstreams", &upstream); err != nil {
+		return nil, fmt.Errorf("get defaults for upstreams: %v", err)
+	}
+	if err := d.Register(&upstream); err != nil {
+		return nil, fmt.Errorf("registering upstream with defaulter: %w", err)
+	}
+
+	target := kong.Target{}
+	if err := d.addEntityDefaults("targets", &target); err != nil {
+		return nil, fmt.Errorf("get defaults for targets: %v", err)
+	}
+	if err := d.Register(&target); err != nil {
+		return nil, fmt.Errorf("registering target with defaulter: %w", err)
+	}
+	return d, nil
+}
+
+func GetDefaulter(ctx context.Context, client *kong.Client) (*Defaulter, error) {
```

Comment:
Same function signature for two exported functions (this and WithClient), remove one.

Reply:
ops, fixed